azure-templates/steps-diff.yml

```diff
@@ -0,0 +1,48 @@
+parameters:
+  - name: lvVersionToDiff
+    type: string
+
+# Note: the following variables are defined in pre-job-steps.yml:
+  # customDeviceRepoName
+
+steps:
+  - powershell: |
+      pip install virtualenv
+      virtualenv env
+      .\env\Scripts\activate
+      pip install requests
+    displayName: Prepare diff environment
+  - checkout: self
+    fetchDepth: 0
+    displayName: Fetch full repo
+  - powershell: |
+      git clone https://github.com/ni/niveristand-custom-device-build-tools.git `
+      $(customDeviceRepoName)\niveristand-custom-device-build-tools
+    displayName: Check out NI VS Custom Device Build Tools repository
+  - powershell: |
+      .\env\Scripts\activate
+      cd $(customDeviceRepoName)
+      $currentDir = Get-Location
+      If (Test-Path $currentDir/diff_dir)
+      {
+        Remove-Item $currentDir/diff_dir -Recurse
+      }
+      New-Item -Path $currentDir -Name "diff_dir" -ItemType "directory"
+      $targetBranch = "$(System.PullRequest.TargetBranch)"
+      $resourcesDir = "niveristand-custom-device-build-tools/resources"
+      python -u $resourcesDir/labview_diff.py `
+      "$currentDir" "$currentDir/diff_dir" "${{parameters.lvVersionToDiff}}" `
+      --target "origin/$targetBranch"
+      $pullRequestID = "$(System.PullRequest.PullRequestNumber)"
+      $jobName = "ni/$(customDeviceRepoName)/PR-$pullRequestID"
+      python -u "$resourcesDir/github_commenter.py" --token "$(diffbot-token)" `
+      --pic-dir "$currentDir/diff_dir" --pull-req "$pullRequestID" --info "$jobName" `
+      --pic-repo "niveristand-diff-bot/diff-images"
+    displayName: Diff VIs and publish
+  - powershell: |
```

Comment:
I have preferred the syntax ` - task: PowerShell@2:` over `- powershell:` because it allows you to define the displayName before the script, which is useful for longer scripts.  I have been pondering just putting these powershell scripts into their own individual files rather than inline, so that they can be reused (and so that any IDE formatting for PowerShell can be more easily leveraged), which would eliminate this though.

Reply:
Implemented the change to `- task: PowerShell@2` in `steps-diff.yml` in https://github.com/ni/niveristand-custom-device-build-tools/pull/164/commits/5cc8a71c74367bbdc2a5e541f82e845ecc05d079. For now, leaving the `powershell` step in `stages.yml`.